tools/test/collect_cc_coverage.sh

```diff
@@ -125,10 +125,24 @@ function gcov_coverage() {
           # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=84879).
           "${GCOV}" -i $COVERAGE_GCOV_OPTIONS -o "$(dirname ${gcda})" "${gcda}"
 
-          # Append all .gcov files in the current directory to the output file.
-          cat *.gcov >> "$output_file"
-          # Delete the .gcov files.
-          rm *.gcov
+          # Extract Gcov's version: the output of `gcov --version` contains the
+          # version as a set of major-minor-patch numbers, of which we extract
+          # the major version.
+          gcov_major_version=$("${GCOV}" --version | sed -n -E -e 's/^.*\s([0-9]+)\.[0-9]+\.[0-9]+\s?.*$/\1/p')
+
+          # Check the gcov version so we can process the data correctly, GCC 9
+          # contains the required changes
+          if [[ $gcov_major_version -ge 9 ]]; then
+              # GCOV 9.1.0 or higher generate a JSON based coverage format
```

Comment:
there is no GCOV 9.0.0?

Reply:
GCC's versioning is a bit weird: 9.0.0 are development releases and the first actual release is 9.1.0. So there are only GCC 9.0.0 development snapshots, which no one (still) uses and even if they are in use, they'll probably already contain this JSON code. 

I probably should have updated the comment to say GCOV 9 instead

Source: https://gcc.gnu.org/develop.html#num_scheme

Reply:
I've clarified the comment a few days ago, is that better?

Reply:
Sorry for the delay. It's stuck in internal review for a while now. Let me see if I can do something about it.